custom_components/keymaster/__init__.py

```diff
@@ -35,327 +46,331 @@
+            ATTR_USER_CODE: usercode,
+        }
+
+        if _using_ozw(hass):
+            servicedata[ATTR_ENTITY_ID] = entity_id
+
             try:
                 await hass.services.async_call(OZW_DOMAIN, SET_USERCODE, servicedata)
             except Exception as err:
                 _LOGGER.error(
                     "Error calling ozw.set_usercode service call: %s", str(err)
                 )
-                pass
+                return
+
+        elif _using_zwave(hass):
+            node_id = _get_node_id(hass, entity_id)
+            if node_id is None:
+                return
+
+            servicedata[ATTR_NODE_ID] = node_id
 
-        else:
-            servicedata = {
-                ATTR_NODE_ID: data,
-                ATTR_CODE_SLOT: code_slot,
-                ATTR_USER_CODE: usercode,
-            }
             try:
                 await hass.services.async_call(ZWAVE_DOMAIN, SET_USERCODE, servicedata)
             except Exception as err:
                 _LOGGER.error(
                     "Error calling lock.set_usercode service call: %s", str(err)
                 )
-                pass
+                return
 
+        else:
+            raise ZWaveIntegrationNotConfiguredError
+
+        # Trigger refresh of user code data
+        await coordinator.async_request_refresh()
         _LOGGER.debug("Add code call completed.")
 
     async def _clear_code(service):
         """Generate the package files"""
         _LOGGER.debug("Clear Code service: %s", service)
         entity_id = service.data[ATTR_ENTITY_ID]
         code_slot = service.data[ATTR_CODE_SLOT]
-        using_ozw = False  # Set false by default
-        if OZW_DOMAIN in hass.data:
-            using_ozw = True  # Set true if we find ozw
-        data = None
-
-        # Pull the node_id from the entity
-        test = hass.states.get(entity_id)
-        if test is not None:
-            data = test.attributes[ATTR_NODE_ID]
-
-        # Bail out if no node_id could be extracted
-        if data is None:
-            _LOGGER.error("Problem pulling node_id from entity.")
-            return
 
         _LOGGER.debug("Attempting to call clear_usercode...")
 
-        if using_ozw:
+        servicedata = {ATTR_CODE_SLOT: code_slot}
+
+        if _using_ozw(hass):
+            # workaround to call dummy slot
             servicedata = {
                 ATTR_ENTITY_ID: entity_id,
                 ATTR_CODE_SLOT: 999,
             }
+
             try:
                 await hass.services.async_call(OZW_DOMAIN, CLEAR_USERCODE, servicedata)
             except Exception as err:
                 _LOGGER.error(
                     "Error calling ozw.clear_usercode service call: %s", str(err)
                 )
-                # pass
 
-            servicedata = {
-                ATTR_ENTITY_ID: entity_id,
-                ATTR_CODE_SLOT: code_slot,
-            }
+            servicedata[ATTR_ENTITY_ID] = entity_id
```

Comment:
I think this actually needs to be the code slot.

Reply:
@firstof9 I’m not sure what you mean

Reply:
I see the problem, pushing a fix shortly